pkg/ingester/ingester.go

```diff
@@ -689,6 +732,10 @@ func (i *Ingester) AllUserStats(ctx context.Context, req *client.UserStatsReques
 
 // Check implements the grpc healthcheck
 func (i *Ingester) Check(ctx context.Context, req *grpc_health_v1.HealthCheckRequest) (*grpc_health_v1.HealthCheckResponse, error) {
+	if err := i.checkRunningOrStopping(); err != nil {
+		return nil, err
```

Comment:
Instead of returning error, have you considered returning a gRPC `HealthCheckResponse` with `HealthCheckResponse_NOT_SERVING` status? I'm not sure which is better, so I'm asking :)

Reply:
I think it doesn't matter too much, as [our code](https://github.com/cortexproject/cortex/blob/9a40de2cafbc20cb324b1f4bb1313c9f50cba0af/pkg/ingester/client/pool.go#L193) only checks for `HealthCheckResponse_SERVING`, but `HealthCheckResponse_NOT_SERVING` may still be nicer, so I'll use that.